test/test_literal.py

```diff
@@ -0,0 +1,165 @@
+from dataclasses import dataclass
+
+from .testutils import *
+
+
+# Not using simple_attribute because floats are disallowed in Literals
+# and booleans don't make sense in this context
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 8), reason="Literal is only supported in Python >= 3.8"
+)
+def test_literal_argument_str():
```

Comment:
Here you should instead take advantage of `SomeClass` being an inner class in order to do something really powerful:


```python3
@pytest.mark.parametrize(
    "allowed_values",
    [
        [123, 456],
        [123],
        ["a", "b"],
        [True, False],
        # etc etc
)
def test_literal_argument(allowed_values: List[Any]):
    @dataclass
    class SomeClass(TestSetup):
        a: Literal[*allowed_values]
    parser = ArgumentParser()
    parser.add_arguments(SomeClass, dest="some_class")
    for allowed_value in allowed_values:
        args = parser.parse_args(shlex.split(f"--a {allowed_value}"))
        assert args.some_class.a == allowed_value
```

Or something like this. Basically what I'm saying is, type annotations are funky, and you can parametrize them (that's kind-of what I'm doing in the above test with `simple_type`)

Reply:
I would but parameterizing Literal with anything other than a literal is not allowed by the PEP. We can do this, but then the test itself would not be PEP-compliant. I also seem to recall this didn't work for some reason.

What would you prefer?

Reply:
I don't mind the test not being pep-compliant. It's not "real" code anyways, it's more like "meta-code" anyways, code generating what would be pep-compliant code! :)
I think this should work, I've done similar kind of hacks with the Union type elsewhere. Let me know if it doesn't though.